homeworks/hw1[345]/shared/src/models.rs

```diff
@@ -0,0 +1,167 @@
+use crate::constants::MAX_CHUNK_BYTE_SIZE;
+use crate::errors::ProcessingErrorsResult;
+use bank_engine::bank::{Operation, TransactionId};
+use serde::{Deserialize, Serialize};
+use std::io::{Read, Write};
+use std::net::TcpStream;
+
+#[derive(Serialize, Debug, Deserialize)]
+pub struct Request {
+    pub payload: RequestPayload,
+}
+
+impl Request {
+    pub fn send(&self, stream: &mut TcpStream) -> Result<(), std::io::Error> {
+        let json = serde_json::to_vec(&self)?;
+        stream.write_all(&json)?;
+        Ok(())
+    }
+}
+
+#[derive(Serialize, Debug, Deserialize)]
+pub enum RequestPayload {
+    /// Represents a ping request.
+    Ping,
+
+    /// Represents an open account request with the specified parameters.
+    OpenAccount(OpenAccountRequestParams),
+
+    /// Represents a withdrawal request with the specified parameters.
+    Withdraw(WithdrawParams),
+
+    /// Represents a deposit request with the specified parameters.
+    Deposit(DepositParams),
+
+    /// Represents a get balance request with the specified parameters.
+    GetBalance(GetBalanceAccountRequestParams),
+
+    /// Represents a transfer request with the specified parameters.
+    Transfer(TransferParams),
+
+    /// Represents a close connection request.
+    CloseConnection,
+
+    /// Represents a get history request without any parameters.
+    GetHistory(),
+
+    /// Represents a get history for account request with the specified account identifier.
+    GetHistoryForAccount(String),
```

Comment:
Было бы прикольно, но и так норм

```suggestion
    GetHistoryForAccount(AccountId),
```

И где-нибудь в `bank.rs`:
```rust
pub type AccountId = String;
```